addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3087,13 +3087,17 @@ const SnippetOptionWidget = Widget.extend({
      * @returns {Promise|undefined}
      */
     selectClass: function (previewMode, widgetValue, params) {
+        const $target = params.target ? this.options.wysiwyg.getEditable().find(params.target) : this.$target;
```

Comment:
It is not clear to me why this change is suddenly needed new. Same with the other methods below which have the same change. Did you *add* the concept of `params.target` or was it existing before but not properly supported in these particular functions ?

Reply:
`this.$target` is defined for the whole snippet option. Here the option is the whole design tab so I needed a way to target a specific element from within the button. This makes sense to me since you might want an option that has a couple of buttons which each target a different element. Why not...?